command/apply/internal/match/match.go

```diff
@@ -105,9 +105,24 @@ func Site(home string, site config.Site, container types.ContainerJSON, blackfir
 		return false
 	}
 
-	// check the path
-	if len(container.Mounts) > 0 {
-		if path != container.Mounts[0].Source {
+	// check the bind mounts for the site
+	mounts, err := site.GetBindMounts(home)
+	if err != nil {
+		return false
+	}
+
+	// if there are more than 1 mount - the site is using excludes
+	switch len(mounts) == 1 {
+	case true:
+		for _, mount := range container.Mounts {
+			// make sure the only bind mount matches the path
+			if (mount.Type == "bind") && path != mount.Source {
+				return false
+			}
+		}
+	default:
+		// check the number of binds matches the number of container binds (we exclude the user home, certs, and nginx configs since they are volumes)
+		if len(mounts) != len(container.Mounts)-3 {
```

Comment:
Instead of hard-coding the 3 container binds we're excluding, maybe set those in an array and check the length to make it a bit more clear.

Reply:
Resolved with https://github.com/craftcms/nitro/pull/412/commits/99fab3a8a817021a08eb97da64b90eff59302337.